src/python/grpcio_tests/tests/observability/_observability_test.py

```diff
@@ -0,0 +1,341 @@
+_STREAM_STREAM = '/test/StreamStream'
+STREAM_LENGTH = 5
+
+_VALID_CONFIG_TRACING_STATS = """
+{
+    "project_id":"test-project",
+    "cloud_trace":{
+       "sampling_rate":1.00
+    },
+    "cloud_monitoring":{},
+}
+"""
+
+# Depends on grpc_core::IsTransportSuppliesClientLatencyEnabled,
+# the following metrcis might not exist.
+_SKIP_VEFIRY = [grpc_observability.MetricsName.CLIENT_TRANSPORT_LATENCY]
+_SPAN_PREFIXS = ['Recv', 'Sent', 'Attempt']
+
+
+class TestExporter(grpc_observability.Exporter):
+
+    def __init__(self, metrics: List[grpc_observability.StatsData],
+                 spans: List[grpc_observability.TracingData]):
+        self.span_collecter = spans
+        self.metric_collecter = metrics
+        self._server = None
+
+    def export_stats_data(
+            self, stats_data: List[grpc_observability.StatsData]) -> None:
+        self.metric_collecter.extend(stats_data)
+
+    def export_tracing_data(
+            self, tracing_data: List[grpc_observability.TracingData]) -> None:
+        self.span_collecter.extend(tracing_data)
+
+
+def handle_unary_unary(request, servicer_context):
+    return _RESPONSE
+
+
+def handle_unary_stream(request, servicer_context):
+    for _ in range(STREAM_LENGTH):
+        yield _RESPONSE
+
+
+def handle_stream_unary(request_iterator, servicer_context):
+    return _RESPONSE
+
+
+def handle_stream_stream(request_iterator, servicer_context):
+    for request in request_iterator:
+        yield _RESPONSE
+
+
+class _MethodHandler(grpc.RpcMethodHandler):
+
+    def __init__(self, request_streaming, response_streaming):
+        self.request_streaming = request_streaming
+        self.response_streaming = response_streaming
+        self.request_deserializer = None
+        self.response_serializer = None
+        self.unary_unary = None
+        self.unary_stream = None
+        self.stream_unary = None
+        self.stream_stream = None
+        if self.request_streaming and self.response_streaming:
+            self.stream_stream = lambda x, y: handle_stream_stream(x, y)
+        elif self.request_streaming:
+            self.stream_unary = lambda x, y: handle_stream_unary(x, y)
+        elif self.response_streaming:
+            self.unary_stream = lambda x, y: handle_unary_stream(x, y)
+        else:
+            self.unary_unary = lambda x, y: handle_unary_unary(x, y)
+
+
+class _GenericHandler(grpc.GenericRpcHandler):
+
+    def service(self, handler_call_details):
+        if handler_call_details.method == _UNARY_UNARY:
+            return _MethodHandler(False, False)
+        elif handler_call_details.method == _UNARY_STREAM:
+            return _MethodHandler(False, True)
+        elif handler_call_details.method == _STREAM_UNARY:
+            return _MethodHandler(True, False)
+        elif handler_call_details.method == _STREAM_STREAM:
+            return _MethodHandler(True, True)
+        else:
+            return None
+
+
+class ObservabilityTest(unittest.TestCase):
```

Comment:
I checked existing interop tests and looks like we already have most of them covered with following exceptions:
* Test trace sampling rate.
  * Will add this.
* Test use config file over env var.
  * Will add this.
* Test empty config.
  * WiIl add this.
* Logging test.
  * Logging not implemented yet.
* Test custom labels.
  * This will be added together with `OpenCensusExporter` (The plan is add it in next PR).
* Test integration with GCP.
  * I don't think we need this since this will be handled by OC StackDriver extension, which can also be test in interop tests.
* Test resource type and labels.
  * Those will be added by OC StackDriver extension, I need to check if we can verify them before sending to GCP, otherwise I suggest test it in interop tests.

Reply:
Sounds great!